app/controllers/profiles_controller.rb

```diff
@@ -52,10 +52,14 @@ def tooltip
 
   def intro
     return redirect_to profile_path(current_user) if current_user&.profile
+
+    @profile_min_reputation = User::Profile::MIN_REPUTATION
+    @may_create_profile = current_user&.may_create_profile?
   end
 
   def new
     return redirect_to profile_path(current_user) if current_user&.profile
+    return redirect_to action: :intro unless current_user&.may_create_profile?
```

Comment:
Neither of these lines need the `&`, nor does the one on L57 as we already check that we have a current user via L2.